src/models/classes/factoryorder.js

```diff
@@ -3,8 +3,8 @@ class FactoryOrder extends OrderCore
     constructor(products)
     {
         var time = GAME.model.config.orderTransportDuration;
-        var id = GAME.model.orders.length;
+        super(products, time);
 
-        super(products, time, id);
+        this.id = GAME.model.orders.length;
```

Comment:
Assigning `id` over here is fine, but we do need the super call. Like `super(products, GAME.model.config.orderTransportDuration)`?